__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -238,6 +529,63 @@ describe('addresses methods', () => {
       address: wallet.getAddressAtIndex(currentAddress.index + 1),
     });
   });
+
+  it('should get correct addresses for a multisig wallet', async () => {
+    const mshWallet = await generateMultisigWalletHelper({ walletIndex: 0 });
+
+    // We will assume the wallet never received txs, which is to be expected for the addresses test
+    expect(mshWallet.getCurrentAddress().address).toStrictEqual(WALLET_CONSTANTS.multisig.addresses[0]);
+
+    for (let i=0; i < 21; ++i) {
+      expect(mshWallet.getAddressAtIndex(i))
+        .toStrictEqual(WALLET_CONSTANTS.multisig.addresses[i]);
+    }
+  })
+
+  it('should respect the current gap limit', async () => {
+    // Lowering the gap limit for the tests
+    const originalGapLimit = wallet.getGapLimit();
+    wallet.setGapLimit(3);
+
+    // Initializing a wallet under those conditions and initializing the getAllAddresses generator
+    const hWallet = await generateWalletHelper();
+
+    // The maximum index generated on an empty wallet should be 3
+    expect(await getMaximumIndexFromAddressGenerator()).toStrictEqual(3)
+
+    // Send a transaction to the address on index 1 and expect the gap limit to be respected
+    await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(1), 1);
+    expect(await getMaximumIndexFromAddressGenerator()).toStrictEqual(4)
+
+    // Restore the original gap limit
+    wallet.setGapLimit(originalGapLimit);
+
+    /**
+     * Iterates the getAllAddresses generator and finds the highest address index for the wallet
+     * @return {Promise<number>}
+     */
+    async function getMaximumIndexFromAddressGenerator() {
+      const addressGenerator = await hWallet.getAllAddresses();
+
+      // Results variable
+      let maximumIndex = 0;
+      let generatorIsActive = true;
+
+      // Address generation loop
+      while (generatorIsActive) {
+        const genResults = await addressGenerator.next();
+
+        // If there is an address,
+        if (!genResults.done) {
+          maximumIndex = genResults.value.index;
+        } else {
+          generatorIsActive = false;
+        }
+      }
+      return maximumIndex;
```

Comment:
An [example](https://javascript.info/async-iterators-generators) on iterating async generators.

```suggestion
      const addressGenerator = hWallet.getAllAddresses();

      // Results variable
      let maximumIndex = 0;
      for await (const address of addressGenerator) {
        if (address.index > maximumIndex) {
          maximumIndex = address.index;
        }
      }
      return maximumIndex;
```

Reply:
Good improvement! Done on [a0e8fdc](https://github.com/HathorNetwork/hathor-wallet-lib/pull/409/commits/a0e8fdcf4878d0722f6c2c86d4efe593b8801d1a)